neqo-transport/src/connection/tests/migration.rs

```diff
@@ -950,3 +951,43 @@ fn retire_prior_to_migration_success() {
     assert_ne!(get_cid(&dgram), original_cid);
     assert_ne!(get_cid(&dgram), probe_cid);
 }
+
+struct GarbageWriter {}
+
+impl crate::connection::test_internal::FrameWriter for GarbageWriter {
+    fn write_frames(&mut self, builder: &mut PacketBuilder) {
+        builder.encode_varint(u32::MAX);
+    }
+}
+
+/// Test the case that we run out of connection ID and receive an invalid frame
+/// from a new path.
+#[test]
+fn error_on_new_path_with_no_connection_id() {
+    let mut client = default_client();
+    let mut server = default_server();
+    connect_force_idle(&mut client, &mut server);
+
+    let dgram = send_something(&mut server, now());
+    let dgram = change_path(&dgram, DEFAULT_ADDR_V4);
```

Comment:
Do you need to change the path for this datagram?  This could create a valid path that is later available for use when you switch for the final test.

I think that your later calls to process_input on the default path will cause a switch back to v6, which should invalidate the path that this switch creates, but it's interesting that we would forget this path.

Reply:
Yes, you are right. We don't need this.